### PR TITLE
working example

### PR DESCRIPTION
**Pull request by memsharded**
I have added a "build.py", you can just run with the exact commands.

The core idea is that you don't need the ``build()`` method, as you are not building, the artifacts are pre-built. Please tell me if this works. Thanks!

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+except:
+    pass
+
+os.system("conan install")
+os.makedirs("mypkg")
+try:
+    old = os.getcwd()
+    os.chdir("mypkg")
+    os.system("conan package .. --build_folder=../")
+    print os.listdir(".")
```
Comment (shreyasbharath) [inline build.py:14]: This doesn't work with Python 3, I will fix it up.

File: conanfile.py [hunk 2]
```diff
@@ -9,25 +9,8 @@ class TestConan(ConanFile):
...
-                shutil.copy2(file, "build/include/" + os.path.basename(file))
-                
-        lib_files = glob.glob("obj/**/*.a", recursive=True)
-        for file in lib_files:
-            if os.path.isfile(file):
-                shutil.copy2(file, "build/lib/" + os.path.basename(file))
-
     def package(self):
-        self.copy("build")
+        self.copy("*", dst="lib", src="obj/libs")
```
Comment (shreyasbharath) [inline conanfile.py:13]: I have also added copying of header files here like so -

    self.copy("*.h", dst="include", src="code", keep_path=False)

**Comment by shreyasbharath** [comment]
The package creation process now works as I expect, thanks!

But I have a question, if I now upload the package like so by running the following commands -
    conan export myuser/testing
    conan upload Test/0.1@PROJECT/testing --all -r artifactory

This is what I end up with, just the package recipe but not the package itself. It doesn't seem to upload the libs and includes, any ideas why?

image

File: conanfile.py (hunk 2 above)
Comment (memsharded) [reply conanfile.py:13]: Good, but for the headers, that is typical to have nested folders like "core/header.h", "subsystem/other_header.h", you might need to not specify ``keep_path=False``, depends on the layout of your package

**Comment by memsharded** [comment]
This is only up to the creation of the package in user space (user folder).
If you want to create the package, you have to ``conan package_files`` from your package folder. Do you want me to add it to the PR? Or maybe better merge and merge with your changes, then a new PR?

File: conanfile.py (hunk 2 above)
Comment (shreyasbharath) [reply conanfile.py:13]: Yep I see what you mean.

The convention we have used in our project passes all the include paths to the compiler so there are no nested #includes.

**Comment by shreyasbharath** [comment]
I will merge this, then submit my changes and then you can do a new PR if that's okay :)

Give me five minutes.

**Pull request merged**